PlutoHelperAgent/PreferencesWindowController.m

```diff
@@ -0,0 +1,210 @@
+        
+        char const* account2 = [account UTF8String];
+        
+        SecKeychainAttribute attrs[] = {
+            
+            { kSecAccountItemAttr, (UInt32)account.length, (char *)account2 }
+            
+        };
+        
+        SecKeychainAttributeList attributes = { 1, attrs };
+        
+        
+        
+        OSStatus dataSaveStatus = SecKeychainItemModifyAttributesAndData(
+                                                                  
+                                                                  itemRef2,
+                                                                  
+                                                                  &attributes,
+                                                                  
+                                                                  (UInt32)password2.length,
+                                                                  
+                                                                  passwordData
+                                                                  
+                                                                  );
+
+        if (dataSaveStatus != errSecSuccess) {
+            
+            NSLog(@"Update failed: %@", SecCopyErrorMessageString(dataSaveStatus, NULL));
+            
+        }
+        
+    } else {
+        
+        NSLog(@"Read failed: %@", SecCopyErrorMessageString(pwSearchStatus, NULL));
+        
+        OSStatus pwSaveStatus = SecKeychainAddGenericPassword(
+                                                         
+                                                         NULL,        // Use default keychain
+                                                         
+                                                         (UInt32)service2.length,
+                                                         
+                                                         [service2 UTF8String],
+                                                         
+                                                         (UInt32)account.length,
+                                                         
+                                                         [account UTF8String],
+                                                         
+                                                         (UInt32)password2.length,
+                                                         
+                                                         passwordData,
+                                                         
+                                                         NULL         // Uninterested in item reference
+                                                         
+                                                         );
+        
+        
+        if (pwSaveStatus != errSecSuccess) {
+            
+            NSLog(@"Write failed: %@", SecCopyErrorMessageString(pwSaveStatus, NULL));
```

Comment:
probably better written as "Keychain write failed" or "could not write password to keychain: "

Reply:
Done and changed various similar log lines.